src/IPC/PipeRx.cc

```diff
@@ -1,22 +1,52 @@
-#include <iostream>
 #include "IPC.h"
 
+#include <iostream>
+#include <fcntl.h>      /* O_RDONLY */
+#include <unistd.h>     /* read() close() unlink() */
+#include <sys/stat.h>   /* mkfifo() */
+#include <errno.h>  
+#include <string.h>
+
+#define FIFO "./fifoChannel"
+
 void PipeRx::receive(void) {
     this->setupPipeRx();
 
-    this->fileSizeRx();
-
     this->pipeRx();
 }
 
 void PipeRx::setupPipeRx(void) {
-    ;
-}
-
-void PipeRx::fileSizeRx(void) {
-    ;
+    mkfifo(FIFO, 0666);
+    this->fd = open(FIFO, O_RDONLY);
+    if (fd < 0) {
+        throw std::runtime_error(strerror(errno));
+    }
+    std::cout << "Rx opened" << std::endl;
 }
 
 void PipeRx::pipeRx(void) {
-    ;
+    ssize_t count;
+
+    while (1) {
+        count = read(this->fd, this->writeBuf, BUFFERSIZE);
```

Comment:
I recommend you to put writeBuf as a variable only for this function not a class member because if you add some methods in your class and you use them then you will still use memory for your buffer but you just need it here.